File: ReSharper.FSharp/test/data/common/assemblyReaderShim/Function pointer 02 - Indexer overloads.fs
module Module

let singleIndexer = SingleIndexer()
let pointerAndString = PointerAndString()

let _: int = singleIndexer[0n]
let _: int = pointerAndString[0n]
let _: int = pointerAndString[""]

let argTypes = ArgTypes()
let byrefArg = ByrefArg()
let byrefReturn = ByrefReturn()

let _: int = argTypes[0n]
let _: int = byrefArg[0n]
let _: int = byrefReturn[0n]

File: ReSharper.FSharp/test/data/common/assemblyReaderShim/Function pointer 02 - Indexer overloads.cs
public unsafe class SingleIndexer
{
  public int this[delegate*<int, int> f] => 1;
}

public unsafe class PointerAndString
{
  public int this[delegate*<int, int> f] => 1;
  public int this[string s] => 2;
}

public unsafe class ArgTypes
{
  public int this[delegate*<int, int> f] => 1;
  public int this[delegate*<string, string> f] => 2;
}

public unsafe class ByrefArg
{
  public int this[delegate*<int, int> f] => 1;
  public int this[delegate*<ref int, int> f] => 2;
}

public unsafe class ByrefReturn
{
  public int this[delegate*<int, int> f] => 1;
  public int this[delegate*<int, ref int> f] => 2;
}
